rasa/shared/core/domain.py

```diff
@@ -1165,14 +1165,25 @@ def persist_clean(self, filename: Union[Text, Path]) -> None:
         rasa.shared.utils.io.write_text_file(as_yaml, filename)
 
     def as_yaml(self, clean_before_dump: bool = False) -> Text:
+        """Dump the `Domain` object as a YAML string.
+        This function preserves the orders of the keys in the domain.
+
+        Args:
+            clean_before_dump: When set to `True`, this method returns
+                               a version of the domain without internal
+                               information. Defaults to `False`.
+        Returns:
+            A string in YAML format representing the domain.
+        """
+        # setting the `version` key first so that it appears at the top of YAML files
+        # thanks to the `should_preserve_key_order` argument of `dump_obj_as_yaml_to_string`
+        domain_data: Dict[Text, Any] = {
```

Comment:
I'm quite surprised this even works 🤔 it's still a Dict, and the keys order is not preserved 😬 
If tests fails, we could change the `dict` to the `OrderedDict` right away!

Reply:
because they changed it in Python 3.7 \0/ => https://softwaremaniacs.org/blog/2020/02/05/dicts-ordered/